From 9a80d8fbe466601a9e35d36088b06a3f2e083db3 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 16 Feb 2016 14:35:51 +0800
Subject: [PATCH 0495/5242] MLK-12402-2 ARM: dts: imx6sl: add power-domains
 for display domain

commit  bf53985acfe27b884742a508a0e1ed4840021c1e from
https://source.codeaurora.org/external/imx/linux-imx.git

Add power-domains for display domain for imx6sl.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sl.dtsi |   12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sl.dtsi b/arch/arm/boot/dts/imx6sl.dtsi
index 412338d..d0756b0 100644
--- a/arch/arm/boot/dts/imx6sl.dtsi
+++ b/arch/arm/boot/dts/imx6sl.dtsi
@@ -727,8 +727,12 @@
 				interrupts = <0 89 IRQ_TYPE_LEVEL_HIGH>;
 				interrupt-parent = <&intc>;
 				pu-supply = <&reg_pu>;
-				clocks = <&clks IMX6SL_CLK_GPU2D_OVG>,
-					 <&clks IMX6SL_CLK_GPU2D_PODF>;
+				clocks = <&clks IMX6SL_CLK_GPU2D_PODF>, <&clks IMX6SL_CLK_GPU2D_OVG>,
+					<&clks IMX6SL_CLK_IPG>, <&clks IMX6SL_CLK_LCDIF_AXI>,
+					<&clks IMX6SL_CLK_LCDIF_PIX>, <&clks IMX6SL_CLK_EPDC_AXI>,
+					<&clks IMX6SL_CLK_EPDC_PIX>, <&clks IMX6SL_CLK_PXP_AXI>;
+				clock-names = "gpu2d_podf", "gpu2d_ovg", "ipg", "lcd_axi",
+						"lcd_pix", "epdc_axi", "epdc_pix", "pxp_axi";
 				#power-domain-cells = <1>;
 			};
 
@@ -751,6 +755,7 @@
 					<&clks IMX6SL_CLK_DUMMY>,
 					<&clks IMX6SL_CLK_DUMMY>;
 				clock-names = "disp-axi", "csi_mclk", "disp_dcic";
+				power-domains = <&gpc 2>;
 				status = "disabled";
 			};
 
@@ -775,11 +780,13 @@
 			pxp: pxp@20f0000 {
 				reg = <0x020f0000 0x4000>;
 				interrupts = <0 98 IRQ_TYPE_LEVEL_HIGH>;
+				power-domains = <&gpc 2>;
 			};
 
 			epdc: epdc@20f4000 {
 				reg = <0x020f4000 0x4000>;
 				interrupts = <0 97 IRQ_TYPE_LEVEL_HIGH>;
+				power-domains = <&gpc 2>;
 			};
 
 			lcdif: lcdif@20f8000 {
@@ -790,6 +797,7 @@
 					 <&clks IMX6SL_CLK_LCDIF_AXI>,
 					 <&clks IMX6SL_CLK_DUMMY>;
 				clock-names = "pix", "axi", "disp_axi";
+				power-domains = <&gpc 2>;
 				status = "disabled";
 			};
 
-- 
1.7.9.5

